File: java/src/neetcode/problems/ContainsDuplicate.java
package neetcode.problems;

public class ContainsDuplicate {
    private final int[] nums = {1, 2, 3, 4, 4, 5, 6, 7, 8};

    public Boolean solve() {
        for (int i = 0; i < nums.length; i++) {
            for (
                    int j = 0; j < nums.length; j++
            ) {
                if (i != j && nums[i] == nums[j]) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: kt/src/main/kotlin/problems/ContainsDuplicate.kt
package problems

fun containsDuplicate(nums: IntArray): Boolean {
    val seenNumbers = HashSet<Int>()

    for (num in nums) {
        if (seenNumbers.contains(num)) {
            return true
        }
        seenNumbers.add(num)
    }

    return false
}

fun main() {
    println("Contains Duplicate")

    val nums = intArrayOf(1, 1, 1, 3, 3, 4, 3, 2, 4, 2)

    println(containsDuplicate(nums))
}
